### ISSUE TITLE
Standalone Executable?

### ISSUE THREAD

**Issue by VicFic2006** [hongquan/CoBang#19]
Just for the sake of testing, making a a standalone executable([AppImage](https://appimage.org/) or [Pyinstaller](https://www.pyinstaller.org/)) might beneficial. Adding a PPA just for trying is a bit cumbersome. And not everybody has Flatpak setup.

**Comment by hongquan** [comment]
@VicFic2006 Flatpak is not my idea. It is just because in early day, many Fedora guys were willing to test, so I tried to package it as Flatpak. I personally don't use Flatpak, because of some limitations in integration with the system (as mentioned in README).

Deploying in PPA is easier for development, because I don't have to build the dependencies. They are prebuilt and host in the repo. I just leave them to package manager (APT) to install. But to package in a box, like Flatpak, I have to write scripts to build *all* dependencies (quite a lot of effort). I'm scared that if I package CoBang to one standalone executable like AppImage/PyInstaller, I have to suffer that experience again.

So, the final answer is, sorry, no.

**Comment by docop** [comment]
Having it fully portable can be quite usefull. Instead of having to download gig of useless and unknow file and then finally used the soft. Appimage is quite portable and we could just put coBang on the usb stick and be fully running.
Thanks to check on that again..

**Comment by hongquan** [comment]
@docop Actually, this "download gig of useless and unknow file" is what your approach does. With traditional distribution (APT), the app reuses many libraries from your system. But with AppImage, you have to duplicate all those system libraries in the AppImage. If the APT package is only hundreds kB, the AppImage will be tens of MB. That extra bits is exactly what you described as "download gig of useless and unknow file".

**Comment by docop** [comment]
Indeed, but as a portable version, just like those in windows system, only 1 file click and it run. Your app can be usefull as a tool on usb stick , as being as is. As for me, i don't have any flatpack and neither snap.
